### PR TITLE
BB2-407 & BB2-408: Migrate build and deploy Jenkins jobs to CloudBees Core

### PR DESCRIPTION
**Pull request by njdister**
**JIRA Ticket:**

[BB2-407](https://jira.cms.gov/browse/BB2-407)
[BB2-408](https://jira.cms.gov/browse/BB2-408)

**User Story or Bug Summary:**

Migrate the following jobs to CB Core:
- BUILD - Platinum AMI
- BUILD - Blue Button API App
- DEPLOY - BB AMI
- DEPLOY - Canary AMI

### What Does This PR Do?

I tried to strike a balance between re-engineering everything from the ground up and only porting what was strictly necessary.  I largely did not touch the code that Jenkins wraps (terraform, ansible, packer) other than some small adjustments to get things working and standardized.  Here's the summary:
- Defines and automates the build of a docker container suitable for our build and deploy jobs in CB Core
- Defines a kubernetes pod for use in CB Core Jenkinsfiles
- Rewrite of the Platinum AMI job, with ability to pass in regex search filter for gold image
    - Simplification/standardization of the Platinum AMI packer file
- Rewrite of the App AMI job
    - Simplification/standardization of the App AMI packer file
- Rewrite of the Deploy AMI job, removal of fragile terraform plan diff in favor of pause for manual plan review, splitting of post deployment playbook into separate job
- Rewrite of the Canary AMI job, add ability to wait for Canary IP to become online
    - Modification of the Canary terraform to support security group needed for polling from CB Core
- New Post Deploy Playbook job, compatible with both AMI and Canary deploy
    - Comment out deprecated code and remove facts gathering in the post deploy playbook for speed optimization

Additionally, I took the opportunity to update our build and deploy tooling:
- Bumps Python from 2.7 to 3.6.  🎉   No more deprecation warnings!
- Bumps Ansible from 2.4 to 2.7  🎉   2.8 and beyond have issues with these old playbooks, let's hope they get nuked before there is another attempt to upgrade.
- Bumps Terraform from 0.11.11 to 0.11.14.  CBJ was using 0.11.14 anyway, so I guess the .terraform-version file wasn't being respected because reasons. 🤔 
- The latest version of everything else at container build time.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

- Did I break anything that used to work?
- Did I miss porting over some vital feature?
- Do we need the deployment placeholder jobs that simply kick off deployments with the right parameters?  Due to the simplification of jobs and parameters I figured this was no longer necessary.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.

### What Needs to Be Merged and Deployed Before this PR?

There are no prerequisites, but some postrequisites once this PR is merged are:
- Disable all jobs in CBJ
- Switch all jobs in CBC to use the master branch

### Submitter Checklist

I have gone through and verified that...:

* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
* [ ] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: Jenkinsfiles/Jenkinsfile.cbc-deploy-canary-ami [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    stage("Wait For Canary") {
+      steps {
+        script {
+          CANARY_IP = sh(returnStdout: true, script: "cd terraform/${APP_ENV}_canary; terraform output private_ip").trim()
+        }
+
+        sh """
+        curl --silent --insecure --retry 40 --retry-delay 15 --retry-connrefused https://${CANARY_IP}/.well-known/openid-configuration
```
Comment (dtisza1) [inline Jenkinsfiles/Jenkinsfile.cbc-deploy-canary-ami]: @njdister Was wondering if the time settings should be increased here? I think the current settings allow for about 10-minutes. Wondering if this should be increased a little bit? Or we can increase it later, if running in to cases where it takes longer.

**Review by dtisza1 — APPROVED** [review]
This looks good to me!    Also setup, configs and testing you've done in CBC look great!

File: Jenkinsfiles/Jenkinsfile.cbc-deploy-canary-ami (hunk 1 above)
Comment (njdister) [reply Jenkinsfiles/Jenkinsfile.cbc-deploy-canary-ami]: Our deployment timeout in terraform is 20 minutes, no harm in keeping parity with that.

**Review by JFU-GIT — APPROVED** [review]
thanks for the demo, looks good to me.

**Pull request merged**